logviewer/src/main/java/org/togetherjava/logwatcher/views/logs/StreamedView.java

```diff
@@ -0,0 +1,153 @@
+        addClassName("hello-world-view");
+
+        add(new Button("Change Columns", this::onChangeColumns), this.grid);
+
+        this.grid.setOperations(logs::findAll, null, null, null);
+        this.grid.setAddOperationVisible(false);
+        this.grid.setDeleteOperationVisible(false);
+        this.grid.setUpdateOperationVisible(false);
+
+
+        this.grid.getGrid()
+            .setColumns(LogEventsConstants.FIELD_INSTANT, LogEventsConstants.FIELD_LOGGER_NAME,
+                    LogEventsConstants.FIELD_MESSAGE);
+        setInstantFormatter();
+        this.grid.getGrid().getColumns().forEach(c -> c.setAutoWidth(true));
+        this.grid.getGrid().getColumns().forEach(c -> c.setResizable(true));
+        this.grid.getGrid().recalculateColumnWidths();
+        this.grid.getGrid().setColumnReorderingAllowed(true);
+
+        VaadinService.getCurrent().addSessionDestroyListener(this::onDestroy);
+        final UI ui = UI.getCurrent();
+        StreamWatcher.addSubscription(this.uuid, () -> ui.access(this.grid::refreshGrid));
+    }
+
+
+    private void onDestroy(SessionDestroyEvent event) {
+        removeHook();
+    }
+
+    @Override
+    protected void onDetach(DetachEvent detachEvent) {
+        removeHook();
+        super.onDetach(detachEvent);
+    }
+
+    private void removeHook() {
+        StreamWatcher.removeSubscription(this.uuid);
+    }
+
+    private void setInstantFormatter() {
+        final Grid<LogEvent> innerGrid = this.grid.getGrid();
+        final Optional<Grid.Column<LogEvent>> column =
+                Optional.ofNullable(innerGrid.getColumnByKey(LogEventsConstants.FIELD_INSTANT));
+        if (column.isEmpty())
+            return;
+
+        final Grid.Column<LogEvent> instant = column.get();
+        innerGrid.removeColumn(instant);
+
+        final String[] keys =
+                innerGrid.getColumns().stream().map(Grid.Column::getKey).toArray(String[]::new);
+        innerGrid.removeAllColumns();
+
+
+        innerGrid
+            .addColumn(new LocalDateTimeRenderer<>(
+                    event -> LocalDateTime.ofInstant(event.getInstant(), ZoneId.systemDefault()),
+                    DateTimeFormatter.ofPattern("yyy-MM-dd HH:mm:ss.SSS")))
+            .setHeader("Instant")
+            .setComparator(Comparator.comparing(LogEvent::getInstant))
+            .setKey(LogEventsConstants.FIELD_INSTANT);
+
+        innerGrid.addColumns(keys);
+    }
+
+    private void onChangeColumns(ClickEvent<Button> event) {
+        final EnhancedDialog dialog = new EnhancedDialog();
+        dialog.setHeader("Choose the Columns you want to see.");
+        final Set<String> columns = this.grid.getGrid()
+            .getColumns()
+            .stream()
+            .map(Grid.Column::getKey)
+            .collect(Collectors.toSet());
+
+        final List<Checkbox> checkBoxes = new ArrayList<>();
+        for (String s : Arrays.asList(LogEventsConstants.FIELD_INSTANT,
```

Comment:
Please rename `s` to something more meaningful.

Reply:
changed it to `field`